Studio-Lovelies/GG-JointJustice-Unity
# Bugfix/ink script validation ignores variables

## Summary

Narrative script validation currently just reads files line by line and checks if the actions give an error. This does not take into account Ink variables though. An action such as `&PAN_TO_POSITION:1,{panTime}` would give an error because {panTime} is not a number. This PR changes the method of validation so that Ink scripts are compiled before being read so that variables are taken into account.

## Testing instructions

On develop try making a narrative script like:
```
&PAN_TO_POSITION:1,{panTime}
VAR panTime = 2
```
This should give an error when validating.

Now try the same thing on this PR's branch. Now a success message should appear.

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
